datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1208,6 +1201,12 @@ fn ensure_distribution(
                 true
             };
 
+            let add_roundrobin = enable_round_robin
+                // Operator benefits from partitioning (e.g. filter):
+                && (would_benefit && repartition_beneficial_stats)
+                // Unless partitioning doesn't increase the partition count, it is not beneficial:
```

Comment:
I think this means:

```suggestion
                // Unless partitioning increases the partition count, it is not beneficial:
```

Reply:
I think you are right. This is original comment and looks incorrect.